airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -1036,58 +1089,62 @@ def run_grant_dataset_view_access(self,
         :type view_dataset: str
         :param view_table: the table of the view
         :type view_table: str
-        :param source_project: the project of the source dataset. If None,
+        :param project_id: the project of the source dataset. If None,
```

Comment:
Is there any way to set self.project_id?

Reply:
Nope, the one is resolved by `_get_credentials_and_project_id` so I duplicated the logic using the decorator and `self.project_id`. Will fix it 🤦‍♂️